lib/sycamore/sycamore/transforms/table_structure/extract.py

```diff
@@ -0,0 +1,143 @@
+        for i, t in enumerate(tokens):
+            assert isinstance(t["bbox"], BoundingBox)
+            t["bbox"].to_absolute_self(width, height).translate_self(-1 * crop_box[0], -1 * crop_box[1])
+            t["bbox"] = t["bbox"].to_list()
+            t["span_num"] = i
+            t["line_num"] = 0
+            t["block_num"] = 0
+        return tokens
+
+    def extract(self, element: TableElement, doc_image: Image.Image) -> TableElement:
+        # We need a bounding box to be able to do anything.
+        if element.bbox is None:
+            return element
+
+        width, height = doc_image.size
+
+        structure_model = TableTransformerForObjectDetection.from_pretrained(self.model)
+
+        # Crop the image to encompass just the table + some padding.
+        padding = 10
+        crop_box = (
+            element.bbox.x1 * width - padding,
+            element.bbox.y1 * height - padding,
+            element.bbox.x2 * width + padding,
+            element.bbox.y2 * height + padding,
+        )
+
+        cropped_image = doc_image.crop(crop_box).convert("RGB")
+
+        # Shift the token bounding boxes to be relative to the cropped image.
+        if element.tokens is not None:
+            tokens = self._prepare_tokens(element.tokens, crop_box, width, height)
+        else:
+            tokens = []
+
+        # Prepare the image. These magic numbers are from the TableTransformer repository.
+        structure_transform = transforms.Compose(
+            [MaxResize(1000), transforms.ToTensor(), transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])]
+        )
+
+        # Run inference using the model and convert the output to raw "objects" containing bounding boxes and types.
+        pixel_values = structure_transform(cropped_image).unsqueeze(0)
+
+        with torch.no_grad():
+            outputs = structure_model(pixel_values)
+
+        structure_id2label = structure_model.config.id2label
+        structure_id2label[len(structure_id2label)] = "no object"
+
+        objects = table_transformers.outputs_to_objects(outputs, cropped_image.size, structure_id2label)
+
+        # Convert the raw objects to our internal table representation. This involves multiple
+        # phases of postprocessing.
+        table = table_transformers.objects_to_table(objects, tokens)
+
+        # Convert cell bounding boxes to be relative to the original image.
+        for cell in table.cells:
+            if cell.bbox is None:
+                continue
+
+            cell.bbox.translate_self(crop_box[0], crop_box[1]).to_relative_self(width, height)
+
+        element.table = table
+        return element
+
+
+DEFAULT_TABLE_STRUCTURE_EXTRACTOR = TableTransformerStructureExtractor()
+
+
+class ExtractTableStructure(Transform):
+    def __init__(self, child: Node, table_structure_extractor: TableStructureExtractor, **resource_args):
```

Comment:
Just curious, do we need this transform independent of partition.

Reply:
I don't think it's necessary. I just thought it might be useful if you wanted to run table extraction model but were using a different partitioner or something. I can remove it if you feel strongly.

Reply:
no strong opinion here.